testing/src/scenario/context.py

```diff
@@ -365,6 +365,51 @@ def action(
             kwargs["id"] = id
         return _Event(f"{name}_action", action=_Action(name, **kwargs))
 
+    @staticmethod
+    def custom(
+        event: ops.BoundEvent,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Event triggered by a charm library.
+
+        For example, suppose that a library uses a ``DatabaseRequirer`` object
+        to emit a ``DatabaseReadyEvent`` event, with charm code::
+
+            class MyCharm(ops.CharmBase):
+                def __init__(self, framework: ops.Framework):
+                    super().__init__(framework)
+                    self.db = DatabaseRequirer(self, 'db-relation')
+                    framework.observe(self.db.on.ready, self._on_db_ready)
+
+        To emit a `DatabaseReadyEvent` event::
+
+            ctx.run(ctx.on.custom(
+                DatabaseRequirer.on.ready, 1, 2, foo='bar'
+            ), state)
+
+        Custom events do not have access to the Juju context of the originating
+        hook event.
+
+        If the custom event takes more arguments than an :attr:`ops.Handle`,
+        pass these as additional arguments to `custom()`. Any of these
```

Comment:
Adding extra backticks:

```suggestion
        pass these as additional arguments to ``custom()``. Any of these
```